specification/cloudshell/resource-manager/Microsoft.Portal/preview/2020-04-01-preview/CloudShell.json

```diff
@@ -0,0 +1,1068 @@
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ConsoleProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ]
+    },
+    "ConsoleProperties": {
+      "description": "Cloud shell console properties.",
+      "type": "object",
+      "required": [
+        "osType",
+        "provisioningState",
+        "uri"
+      ],
+      "properties": {
+        "osType": {
+          "description": "The operating system type of the cloud shell.",
+          "type": "string",
+          "enum": [
+            "Linux",
+            "Windows"
+          ],
+          "x-ms-enum": {
+            "name": "osType",
+            "modelAsString": true
+          }
+        },
+        "provisioningState": {
+          "description": "Provisioning state of the console.",
+          "type": "string",
+          "enum": [
+            "NotSpecified",
+            "Accepted",
+            "Pending",
+            "Updating",
+            "Creating",
+            "Repairing",
+            "Failed",
+            "Canceled",
+            "Succeeded"
+          ],
+          "x-ms-enum": {
+            "name": "provisioningState",
+            "modelAsString": true
+          }
+        },
+        "uri": {
+          "description": "Uri of the console.",
+          "type": "string"
+        }
+      }
+    },
+    "ConsoleCreateProperties": {
+      "description": "Cloud shell properties for creating a console.",
+      "type": "object",
+      "required": [
+        "osType"
+      ],
+      "properties": {
+        "osType": {
+          "description": "The operating system type of the cloud shell.",
+          "type": "string",
+          "enum": [
+            "Linux",
+            "Windows"
+          ],
+          "x-ms-enum": {
+            "name": "osType",
+            "modelAsString": true
+          }
+        },
+        "provisioningState": {
+          "description": "Provisioning state of the console.",
+          "type": "string",
+          "enum": [
+            "NotSpecified",
+            "Accepted",
+            "Pending",
+            "Updating",
+            "Creating",
+            "Repairing",
+            "Failed",
+            "Canceled",
+            "Succeeded"
+          ],
+          "x-ms-enum": {
+            "name": "provisioningState",
+            "modelAsString": true
+          }
+        },
+        "uri": {
+          "description": "Uri of the console.",
+          "type": "string"
+        }
+      }
+    },
+    "ConsoleDefinition": {
+      "description": "Console definition",
+      "type": "object",
+      "required": [
+        "properties"
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ConsoleCreateProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ]
+    },
+    "Resource": {
```

Comment:
>Resource [](http://example.com/codeflow?start=5&length=8)

Suggest referencing specification/common-types/resource-management/v2/types.json rather than copying the Resource properties locally.